crates/bevy_pbr/src/ssr/mod.rs

```diff
@@ -0,0 +1,415 @@
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    query::{Has, QueryItem, With},
+    reflect::ReflectComponent,
+    schedule::IntoSystemConfigs as _,
+    system::{lifetimeless::Read, Commands, Query, Res, ResMut, Resource},
+    world::{FromWorld, World},
+};
+use bevy_reflect::Reflect;
+use bevy_render::{
+    extract_component::{ExtractComponent, ExtractComponentPlugin},
+    render_graph::{NodeRunError, RenderGraphApp, RenderGraphContext, ViewNode, ViewNodeRunner},
+    render_resource::{
+        binding_types, BindGroupEntries, BindGroupLayout, BindGroupLayoutEntries,
+        CachedRenderPipelineId, ColorTargetState, ColorWrites, DynamicUniformBuffer, FragmentState,
+        Operations, PipelineCache, RenderPassColorAttachment, RenderPassDescriptor,
+        RenderPipelineDescriptor, Shader, ShaderStages, ShaderType, SpecializedRenderPipeline,
+        SpecializedRenderPipelines, TextureFormat, TextureSampleType,
+    },
+    renderer::{RenderContext, RenderDevice, RenderQueue},
+    texture::BevyDefault as _,
+    view::{ExtractedView, Msaa, ViewTarget, ViewUniformOffset},
+    Render, RenderApp, RenderSet,
+};
+use bevy_utils::prelude::default;
+
+use crate::{
+    binding_arrays_are_usable, graph::NodePbr, prelude::EnvironmentMapLight,
+    MeshPipelineViewLayoutKey, MeshPipelineViewLayouts, MeshViewBindGroup, RenderViewLightProbes,
+    ViewFogUniformOffset, ViewLightProbesUniformOffset, ViewLightsUniformOffset,
+};
+
+const SSR_SHADER_HANDLE: Handle<Shader> = Handle::weak_from_u128(10438925299917978850);
+
+/// Enables screen-space reflections for a camera.
+///
+/// Screen-space reflections are currently only supported with deferred rendering.
+pub struct ScreenSpaceReflectionsPlugin;
+
+/// Add this component to a camera to enable *screen-space reflections* (SSR).
+///
+/// Screen-space reflections currently require deferred rendering in order to
+/// appear. Therefore, you'll generally need to add a [`DepthPrepass`] and a
+/// [`DeferredPrepass`] to the camera as well.
+///
+/// SSR currently performs no roughness filtering for glossy reflections, so
+/// only very smooth surfaces will reflect objects in screen space. You can
+/// adjust the `perceptual_roughness_threshold` in order to tune the threshold
+/// below which screen-space reflections will be traced.
+///
+/// As with all screen-space techniques, SSR can only reflect objects on screen.
+/// When objects leave the camera, they will disappear from reflections.
+/// Alternatives that don't suffer from this problem include
+/// [`crate::environment_map::ReflectionProbeBundle`]s. The advantage of SSR is
+/// that it can reflect all objects, not just static ones.
+///
+/// SSR is an approximation technique and produces artifacts in some situations.
+/// Hand-tuning the settings in this component will likely be useful.
+#[derive(Clone, Copy, Component, Reflect, ShaderType)]
+#[reflect(Component)]
+pub struct ScreenSpaceReflections {
```

Comment:
Isn't the convention that these kind of components are called `ScreenSpaceReflectionsSettings`?

Reply:
I did the renaming.